crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2100,8 +2100,10 @@ impl<'db> Class<'db> {
             };
 
             let Type::ClassLiteral(mut candidate) = metaclass else {
-                // If the metaclass is not a class, return it directly.
-                return Ok(metaclass);
+                // TODO: If the metaclass is not a class, we should verify that it's a callable
+                // which accepts the same arguments as `type.__new__` (otherwise error), and return
```

Comment:
And also verify that it returns a type assignable to `Instance(type)`? I don't think we can realistically support `class` statements that don't create classes

Reply:
Honestly it probably wouldn't be that hard, but it's also likely not worth it.